packages/manager/modules/pci/src/projects/project/instances/instance/unshelve/unshelve.controller.js

```diff
@@ -0,0 +1,76 @@
+import get from 'lodash/get';
+
+export default class PciInstanceUnshelveController {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiCloudProjectInstance,
+    PciProjectsProjectInstanceService,
+    Poller,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+    this.PciProjectsProjectInstanceService = PciProjectsProjectInstanceService;
+    this.Poller = Poller;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+  }
+
+  startPollerForUnshelveInstance() {
+    const endPointUrl = `/cloud/project/${this.projectId}/instance/${this.instance.id}`;
+    this.Poller.poll(endPointUrl, null, {
+      interval: 3000,
+      successRule(instance) {
+        return instance.status === 'ACTIVE';
+      },
+      namespace: 'cloud.project.instance.unshelve',
+      notifyOnError: false,
+    }).then(() => {
+      if (this.isInstancesStateScope()) {
```

Comment:
Is this check really required? Can't we call `refreshInstances` if `refreshInstances` is available (by checking `if (this.refreshInstances)...`)? :)

Reply:
@ganeshkumar1989, goals of my test, is to refresh page only if customer is in instances or instances child states to refresh the related page. So i don't need to refresh the page if customer is moved to another state (eg: create a project).

I hope it's clear.

Reply:
Got it. Thanks

Reply:
Thanks @ganeshkumar1989 , i no longer need.